cli/src/pcluster/models/cluster.py

```diff
@@ -523,10 +524,12 @@ def _render_and_upload_byos_template(self):
                 f"Error while downloading scheduler plugin artifacts from '{byos_template}': {str(e)}"
             ) from e
 
-        # TODO: apply jinja rendering before upload
+        # jinja rendering
+        template = Template(file_content)
+        rendered_template = template.render(cluster_configuration=self.config)
```

Comment:
probably better to pass the config as is rather than the one that gets mapped in our internal data structures. probably with something like `parse_config(self.source_config_text)`?